src/sentry/lang/javascript/processor.py

```diff
@@ -341,6 +346,44 @@ def fetch_release_body():
     return result
 
 
+def _get_from_archive(url: str, archive: ReleaseArchive) -> bytes:
+    candidates = ReleaseFile.normalize(url)
+    for candidate in candidates:
+        try:
+            return archive.get_file_by_url(candidate)
+        except KeyError:
+            pass
+
+    # None of the filenames matched
+    raise KeyError(f"Not found in archive: '{url}'")
+
+
+@metrics.wraps("sourcemaps.fetch_release_artifact")
+def fetch_release_artifact(url, release, dist):
+    """
+    Get a release artifact either by extracting it or fetching it directly.
+
+    If a release archive was saved, the individual file will be extracted
+    from the archive.
+
+    """
+    release_file = fetch_release_file(RELEASE_ARCHIVE_FILENAME, release, dist)
+
+    if release_file is not None:
+        zipobj = BytesIO(release_file.body)
```

Comment:
We might optimize this by working with a file descriptor rather than reading all the bytes of the zip file.